gnome-initial-setup/gis-driver.c

```
@@ -288,6 +294,208 @@ gis_driver_get_mode (GisDriver *driver)
+        }
+    }
+}
+
+void
+replace_google_chrome_in_favorite_apps (GSettings *gnome_shell_settings)
+{
+  GStrv favorite_apps = g_settings_get_strv (gnome_shell_settings,
+                                             GNOME_SHELL_FAVORITE_APPS);
+  replace_string_in_strv (favorite_apps,
+                          GOOGLE_CHROME_DESKTOP_ID,
+                          CHROMIUM_BROWSER_DESKTOP_ID);
+  g_settings_set_strv (gnome_shell_settings,
+                       GNOME_SHELL_FAVORITE_APPS,
+                       (const gchar * const *) favorite_apps);
+
+  g_strfreev (favorite_apps);
+}
+
+void
+configure_demo_mode_replace_chrome_with_chromium ()
+{
+  GSettings *gnome_shell_settings = g_settings_new (GNOME_SHELL_SCHEMA);
+  replace_google_chrome_in_favorite_apps (gnome_shell_settings);
+  g_object_unref (gnome_shell_settings);
+}
+
+gboolean
+setup_demo_config (GisDriver *driver, GError **error)
+{
+  gchar *stamp_file = g_build_filename (g_get_user_config_dir (),
+                                        "eos-demo-mode",
+                                        NULL);
+
+  if (!g_file_set_contents (stamp_file, "1", sizeof(char), error))
+    {
+      g_free (stamp_file);
+      return FALSE;
+    }
+
+  configure_demo_mode_power_settings ();
+  configure_demo_mode_disallow_app_center_updates ();
+  configure_demo_mode_replace_chrome_with_chromium ();
+
+  g_free (stamp_file);
+  return TRUE;
+}
+
+void
+gis_driver_enter_demo_mode (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+
+  if (priv->is_in_demo_mode)
+    return;
+
+  /* Enter test mode.
+   *
+   * After this point, all data on the system will become ephemeral,
+   * including /home, which means any changes will be lost on reboot.
+   */
+  GError *error = NULL;
+  if (!gis_pkexec (LIBEXECDIR "/eos-demo-mode", "demo-guest", NULL, &error))
+    {
+      handle_demo_mode_error (error);
+      return;
+    }
+
+  if (!create_demo_user (driver))
```

It looks like this function should also return the error instead of independently calling `hanlde_demo_mode_error()`